### PR TITLE
feat: add validation component

### PR DESCRIPTION
**Pull request by josokinas**
## Purpose

Add Validation component.

## Approach

Implements Validation component.

This closes issue #5.

## Testing

On Storybook preview.

## Risks

N/A

### CODE REVIEW

File: packages/react/src/components/validation/validation.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { c, classy, m } from '@onfido/castor';
+import { Icon } from '@onfido/castor-react';
+import React from 'react';
+
+/**
+ * Intended to be used alongside field components.
+ *
+ * `Validation` uses an `Icon` that requires `Icons` (SVG sprite) to be included
```
Comment (rabelloo) [inline packages/react/src/components/validation/validation.tsx:8]: If you use `showIcon`, we should note that.

File: packages/react/src/components/validation/validation.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ */
+export const Validation = ({
+  showIcon,
+  children,
+  className,
+  ...restProps
+}: ValidationProps): JSX.Element => (
+  <div
+    {...restProps}
+    className={classy(c('validation'), m({ showIcon }), className)}
```
Comment (rabelloo) [inline packages/react/src/components/validation/validation.tsx:21]: I don't think `m({ showIcon })` is doing anything, is it?

File: packages/react/src/components/validation/validation.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  >
+    {showIcon && <Icon name="error" />}
+    {children}
+  </div>
+);
+
+export type ValidationProps = JSX.IntrinsicElements['div'] & {
+  showIcon?: boolean;
```
Comment (rabelloo) [inline packages/react/src/components/validation/validation.tsx:29]: 💅🏻  how do you feel about `withIcon`?

File: packages/core/src/components/validation/validation.scss [hunk 4]
```diff
@@ -0,0 +1,15 @@
+@use '../../helpers';
+
+@mixin Validation() {
+  .ods-validation {
```
Comment (rabelloo) [inline packages/core/src/components/validation/validation.scss:4]: My issue with the name is that it's always a red/negative/error message, even according to its own style.

I'd call it `error` instead, or something that denotes an unsuccessful/invalid state of validation.

Like, you wouldn't use this component for any other kind of message, for example:

```tsx
<Validation success>Your password complexity is safe enough!</Validation>
```

Comment (josokinas) [reply packages/core/src/components/validation/validation.scss:4]: I understand this.

However, component naming should still align both in design and engineering. I will suggest this change to designers.

File: packages/react/src/components/validation/validation.tsx (hunk 2 above)
Comment (josokinas) [reply packages/react/src/components/validation/validation.tsx:21]: Good point, I added but realised no need to do anything with styling.

**Review by rabelloo — APPROVED** [review]

**Pull request merged**